#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.1495332448099039 0.3134123298198807 0.9377699719086209 0.9412851268661754
  position 0.003023422367132267 -0.5110873318658977 0.36783269378951133
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 1.8 1.2
  floorAppearance NULL
}
Solid {
  rotation 0 1 0 0
  scale 0.01 0.01 0.01
  children [
    Solid {
      rotation 0 1 0 0
      name "obj_0"
    }
    Solid {
      rotation 0 1 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
            name "color_2829873"
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                62.864 29.067 0.01, 68.238 28.082 0.01, 65.702 31.172 0.01, -48.971 -42.747 0, -48.971 -42.747 0.01, -45.778 -44.258 0.01, -45.778 -44.258 0, 65.085 25.743 0, 65.085 25.743 0.01, 61.718 23.725 0, 61.718 23.725 0.01, 81 47 0, 79.817 43.695 0, 76.7592 47 0, 76.124 45.224 0, 76.121 36.779 0, 72.797 39 0, 78.139 40.146 0, -52.002 -40.931 0, -52.002 -40.931 0.01, 74.613 42.031 0, -54.84 -38.826 0, -54.84 -38.826 0.01, -44.003 -44.8929 0, -44.003 -49.1337 0.01, -44.003 -49.1337 0, 73.782 33.626 0, 70.692 36.162 0, 68.32 33.544 0, -71.807 -7.075 0, -72 -10.996 0, -72 -10.996 0.01, -71.807 -7.075 0.01, 65.702 31.172 0, 68.238 28.082 0, 71.146 30.718 0, -71.807 -14.917 0, -71.807 -14.917 0.01, -71.231 -18.8 0, -71.231 -18.8 0.01, 62.864 29.067 0, -70.278 -22.607 0, -70.278 -22.607 0.01, 59.833 27.251 0.01, -68.955 -26.303 0, -68.955 -26.303 0.01, 56.64 25.74 0.01, -67.277 -29.852 0, -67.277 -29.852 0.01, 53.313 24.55 0.01, 58.169 22.047 0.01, 73.782 33.626 0.01, 68.32 33.544 0.01, 71.146 30.718 0.01, 49.886 23.691 0.01, 54.473 20.724 0.01, -65.259 -33.219 0, -65.259 -33.219 0.01, 46.391 23.173 0.01, 46.783 19.195 0.01, 42.862 23 0.01, 42.862 19.002 0.01, -62.92 -36.372 0, -62.92 -36.372 0.01, -60.284 -39.28 0, -60.284 -39.28 0.01, 50.666 19.771 0.01, -57.376 -41.916 0, -57.376 -41.916 0.01, 59.833 27.251 0, 56.64 25.74 0, -57.458 -36.454 0, -57.458 -36.454 0.01, 58.169 22.047 0, 53.313 24.55 0, 54.473 20.724 0, -59.83 -33.836 0, -59.83 -33.836 0.01, 49.886 23.691 0, 50.666 19.771 0, -61.935 -30.998 0, -61.935 -30.998 0.01, -63.751 -27.967 0, -63.751 -27.967 0.01, 46.783 19.195 0, 46.391 23.173 0, 42.862 19.002 0, 42.862 23 0, -65.262 -24.774 0, -65.262 -24.774 0.01, -66.452 -21.447 0, -66.452 -21.447 0.01, -67.311 -18.02 0, -67.311 -18.02 0.01, -67.829 -14.525 0, -67.829 -14.525 0.01, -68.003 -10.996 0, -68.003 -10.996 0.01, 39.333 23.173 0, -67.829 -7.467 0, -67.829 -7.467 0.01, -25 24.002 0, -25 20.002 0, -25 20.002 0.01, -25 24.002 0.01, 17 24.002 0.01, 17 20.002 0.01, 17 20.002 0, 17 24.002 0, -66.257 7.225 0.01, -62.933 5.004 0.01, -63.918 10.378 0.01, -58.456 10.46 0.01, -61.282 13.286 0.01, -68.275 3.858 0.01, -58.374 15.922 0.01, -55.838 12.832 0.01, -53 14.937 0.01, -46.776 18.264 0.01, -51.854 20.279 0.01, -49.969 16.753 0.01, -43.449 19.454 0.01, -48.305 21.957 0.01, -47.307 -47.951 0.01, -44.003 -44.8929 0.01, 25 24.002 0, 34 24.002 0, 34 20.002 0, 25 20.002 0, -50.856 -46.273 0.01, 34.864 19.8196 0, 34.864 23.9351 0, 35.838 23.691 0, 38.941 19.195 0, 35.058 19.771 0, 76.124 45.224 0.01, 79.817 43.695 0.01, 76.7592 47 0.01, 74.613 42.031 0.01, 78.139 40.146 0.01, 76.121 36.779 0.01, 72.797 39 0.01, 81 47 0.01, 70.692 36.162 0.01, -54.223 -44.255 0.01, 39.333 23.173 0.01, 35.838 23.691 0.01, 38.941 19.195 0.01, 35.058 19.771 0.01, 34.864 19.8196 0.01, 34.864 23.9351 0.01, -25.001 24.1847 0.01, -25.001 24.1847 0, -25.194 24.233 0, -25.194 24.233 0.01, -29.077 24.809 0, -29.077 24.809 0.01, -32.998 25.002 0, -32.998 25.002 0.01, -36.919 24.809 0, -36.919 24.809 0.01, -40.802 24.233 0, -40.802 24.233 0.01, -44.609 23.28 0, -44.609 23.28 0.01, -48.305 21.957 0, 34 20.002 0.01, 34 24.002 0.01, 25 20.002 0.01, 25 24.002 0.01, -51.854 20.279 0, -55.221 18.261 0, -55.221 18.261 0.01, -49.969 16.753 0, -53 14.937 0, -46.776 18.264 0, -43.449 19.454 0, -40.022 20.313 0, -40.022 20.313 0.01, -36.527 20.831 0, -36.527 20.831 0.01, -32.998 21.005 0, -32.998 21.005 0.01, -29.469 20.831 0, -29.469 20.831 0.01, -25.974 20.313 0, -25.974 20.313 0.01, -25.001 20.0691 0, -25.001 20.0691 0.01, -58.374 15.922 0, -61.282 13.286 0, -63.918 10.378 0, -47.307 -47.951 0, -66.257 7.225 0, -68.275 3.858 0, -50.856 -46.273 0, -69.953 0.309 0.01, -54.223 -44.255 0, -67.311 -3.972 0, -69.953 0.309 0, -71.1824 -2.998 0.01, -71.1824 -2.998 0, -71.231 -3.192 0, -71.231 -3.192 0.01, -71.1357 -2.995 0, -71.1357 -2.995 0.01, -67.311 -3.972 0.01, -66.26 -1.22 0, -66.26 -1.22 0.01, -66.8949 -2.995 0.01, -66.8949 -2.995 0, -67.0669 -2.998 0, -67.0669 -2.998 0.01, -64.749 1.973 0, -64.749 1.973 0.01, -62.933 5.004 0, -60.828 7.842 0, -60.828 7.842 0.01, -58.456 10.46 0, -55.838 12.832 0
              ]
            }
            coordIndex [
              0, 1, 2, -1, 3, 4, 5, -1, 3, 5, 6, -1, 7, 8, 9, -1, 9, 8, 10, -1, 11, 12, 13, -1, 14, 13, 12, -1, 15, 16, 17, -1, 18, 19, 4, -1, 18, 4, 3, -1, 17, 14, 12, -1, 14, 17, 20, -1, 21, 22, 19, -1, 21, 19, 18, -1, 23, 24, 25, -1, 20, 17, 16, -1, 15, 26, 16, -1, 27, 16, 26, -1, 28, 27, 26, -1, 29, 30, 31, -1, 29, 31, 32, -1, 33, 28, 34, -1, 35, 34, 28, -1, 26, 35, 28, -1, 31, 30, 36, -1, 31, 36, 37, -1, 36, 38, 39, -1, 36, 39, 37, -1, 40, 33, 34, -1, 38, 41, 42, -1, 38, 42, 39, -1, 43, 10, 0, -1, 41, 44, 45, -1, 41, 45, 42, -1, 46, 10, 43, -1, 44, 47, 48, -1, 44, 48, 45, -1, 49, 50, 46, -1, 51, 52, 53, -1, 54, 55, 49, -1, 53, 52, 1, -1, 47, 56, 57, -1, 47, 57, 48, -1, 58, 59, 54, -1, 1, 0, 8, -1, 60, 61, 58, -1, 8, 0, 10, -1, 56, 62, 63, -1, 56, 63, 57, -1, 62, 64, 65, -1, 62, 65, 63, -1, 66, 54, 59, -1, 55, 54, 66, -1, 50, 49, 55, -1, 10, 46, 50, -1, 59, 58, 61, -1, 67, 68, 65, -1, 67, 65, 64, -1, 34, 7, 40, -1, 7, 9, 40, -1, 69, 40, 9, -1, 70, 69, 9, -1, 71, 72, 21, -1, 22, 21, 72, -1, 9, 73, 70, -1, 74, 70, 73, -1, 73, 75, 74, -1, 76, 77, 72, -1, 76, 72, 71, -1, 78, 74, 75, -1, 75, 79, 78, -1, 80, 81, 77, -1, 80, 77, 76, -1, 82, 83, 81, -1, 82, 81, 80, -1, 79, 84, 78, -1, 85, 78, 84, -1, 84, 86, 85, -1, 87, 85, 86, -1, 88, 89, 83, -1, 88, 83, 82, -1, 90, 91, 89, -1, 90, 89, 88, -1, 92, 93, 91, -1, 92, 91, 90, -1, 94, 95, 93, -1, 94, 93, 92, -1, 95, 94, 96, -1, 95, 96, 97, -1, 87, 86, 98, -1, 99, 100, 97, -1, 99, 97, 96, -1, 101, 102, 103, -1, 104, 101, 103, -1, 105, 104, 103, -1, 105, 103, 106, -1, 107, 102, 101, -1, 107, 101, 108, -1, 103, 102, 107, -1, 103, 107, 106, -1, 105, 108, 101, -1, 105, 101, 104, -1, 109, 110, 111, -1, 111, 112, 113, -1, 114, 110, 109, -1, 113, 112, 115, -1, 116, 117, 115, -1, 118, 119, 120, -1, 121, 122, 118, -1, 5, 123, 124, -1, 125, 126, 127, -1, 128, 125, 127, -1, 4, 129, 5, -1, 19, 129, 4, -1, 130, 131, 132, -1, 86, 133, 98, -1, 132, 98, 133, -1, 134, 132, 133, -1, 130, 132, 134, -1, 22, 68, 19, -1, 129, 123, 5, -1, 135, 136, 137, -1, 138, 139, 135, -1, 140, 141, 51, -1, 139, 141, 140, -1, 136, 135, 139, -1, 142, 137, 136, -1, 24, 124, 123, -1, 77, 63, 72, -1, 72, 68, 22, -1, 143, 51, 141, -1, 81, 63, 77, -1, 68, 144, 19, -1, 52, 51, 143, -1, 141, 139, 138, -1, 83, 48, 81, -1, 129, 19, 144, -1, 2, 1, 52, -1, 89, 48, 83, -1, 91, 45, 89, -1, 68, 72, 65, -1, 39, 42, 93, -1, 65, 72, 63, -1, 95, 97, 37, -1, 63, 81, 57, -1, 57, 81, 48, -1, 145, 146, 147, -1, 148, 146, 149, -1, 31, 37, 97, -1, 97, 100, 32, -1, 32, 31, 97, -1, 93, 95, 37, -1, 37, 39, 93, -1, 91, 93, 42, -1, 45, 91, 42, -1, 48, 89, 45, -1, 150, 149, 146, -1, 151, 152, 153, -1, 151, 153, 154, -1, 153, 155, 156, -1, 153, 156, 154, -1, 155, 157, 158, -1, 155, 158, 156, -1, 157, 159, 160, -1, 157, 160, 158, -1, 159, 161, 162, -1, 159, 162, 160, -1, 161, 163, 164, -1, 161, 164, 162, -1, 163, 165, 122, -1, 163, 122, 164, -1, 166, 167, 168, -1, 169, 168, 167, -1, 165, 170, 119, -1, 165, 119, 122, -1, 170, 171, 172, -1, 170, 172, 119, -1, 60, 145, 61, -1, 61, 145, 147, -1, 147, 146, 148, -1, 115, 172, 171, -1, 15, 140, 51, -1, 15, 17, 139, -1, 15, 139, 140, -1, 17, 136, 139, -1, 173, 120, 117, -1, 173, 117, 174, -1, 136, 17, 12, -1, 12, 11, 142, -1, 12, 142, 136, -1, 175, 118, 120, -1, 175, 120, 173, -1, 14, 135, 137, -1, 14, 137, 13, -1, 176, 121, 118, -1, 176, 118, 175, -1, 177, 178, 121, -1, 177, 121, 176, -1, 179, 180, 178, -1, 179, 178, 177, -1, 142, 11, 137, -1, 13, 137, 11, -1, 131, 130, 150, -1, 181, 182, 180, -1, 181, 180, 179, -1, 126, 125, 167, -1, 183, 184, 182, -1, 183, 182, 181, -1, 84, 59, 61, -1, 84, 61, 86, -1, 79, 66, 59, -1, 79, 59, 84, -1, 185, 186, 184, -1, 185, 184, 183, -1, 167, 125, 169, -1, 75, 55, 66, -1, 75, 66, 79, -1, 187, 188, 186, -1, 187, 186, 185, -1, 125, 128, 169, -1, 73, 50, 55, -1, 73, 55, 75, -1, 9, 10, 73, -1, 151, 188, 152, -1, 187, 152, 188, -1, 73, 10, 50, -1, 126, 167, 127, -1, 171, 189, 115, -1, 149, 150, 130, -1, 189, 190, 113, -1, 189, 113, 115, -1, 106, 107, 108, -1, 106, 108, 105, -1, 190, 191, 111, -1, 190, 111, 113, -1, 25, 192, 23, -1, 6, 23, 192, -1, 191, 193, 109, -1, 191, 109, 111, -1, 193, 194, 114, -1, 193, 114, 109, -1, 3, 6, 195, -1, 114, 194, 196, -1, 18, 3, 195, -1, 195, 197, 18, -1, 197, 67, 18, -1, 21, 18, 67, -1, 171, 170, 174, -1, 195, 6, 192, -1, 173, 174, 170, -1, 175, 173, 170, -1, 170, 165, 175, -1, 176, 175, 165, -1, 71, 21, 67, -1, 165, 163, 176, -1, 177, 176, 163, -1, 181, 155, 183, -1, 67, 64, 71, -1, 64, 62, 71, -1, 183, 155, 185, -1, 76, 71, 62, -1, 80, 76, 62, -1, 62, 56, 80, -1, 56, 47, 80, -1, 82, 80, 47, -1, 88, 82, 47, -1, 187, 185, 152, -1, 163, 161, 177, -1, 161, 159, 177, -1, 179, 177, 159, -1, 181, 179, 159, -1, 159, 157, 181, -1, 155, 181, 157, -1, 153, 185, 155, -1, 152, 185, 153, -1, 47, 44, 88, -1, 90, 88, 44, -1, 44, 41, 90, -1, 92, 90, 41, -1, 96, 94, 36, -1, 99, 29, 198, -1, 41, 38, 92, -1, 38, 36, 92, -1, 94, 92, 36, -1, 36, 30, 96, -1, 30, 29, 96, -1, 99, 96, 29, -1, 196, 194, 199, -1, 200, 201, 202, -1, 200, 202, 203, -1, 120, 119, 117, -1, 199, 204, 205, -1, 199, 205, 196, -1, 99, 198, 100, -1, 206, 100, 198, -1, 180, 160, 178, -1, 178, 164, 121, -1, 115, 117, 172, -1, 207, 208, 209, -1, 207, 209, 210, -1, 182, 160, 180, -1, 172, 117, 119, -1, 151, 154, 186, -1, 211, 212, 206, -1, 211, 206, 198, -1, 188, 151, 186, -1, 162, 178, 160, -1, 164, 178, 162, -1, 122, 121, 164, -1, 119, 118, 122, -1, 158, 160, 182, -1, 184, 156, 182, -1, 154, 156, 186, -1, 184, 186, 156, -1, 158, 182, 156, -1, 213, 214, 208, -1, 213, 208, 207, -1, 100, 206, 32, -1, 215, 110, 214, -1, 215, 214, 213, -1, 201, 212, 211, -1, 200, 203, 206, -1, 32, 206, 203, -1, 205, 210, 209, -1, 214, 114, 208, -1, 216, 217, 110, -1, 216, 110, 215, -1, 212, 201, 200, -1, 212, 200, 206, -1, 218, 112, 217, -1, 218, 217, 216, -1, 208, 196, 209, -1, 205, 209, 196, -1, 110, 114, 214, -1, 219, 116, 112, -1, 219, 112, 218, -1, 174, 117, 116, -1, 174, 116, 219, -1, 217, 111, 110, -1, 32, 203, 29, -1, 202, 29, 203, -1, 112, 111, 217, -1, 116, 115, 112, -1, 196, 208, 114, -1, 210, 205, 204, -1, 68, 67, 144, -1, 197, 144, 67, -1, 195, 129, 144, -1, 195, 144, 197, -1, 129, 195, 123, -1, 192, 123, 195, -1, 25, 24, 123, -1, 25, 123, 192, -1, 6, 5, 124, -1, 6, 124, 23, -1, 23, 124, 24, -1, 29, 202, 198, -1, 34, 1, 8, -1, 34, 8, 7, -1, 210, 199, 207, -1, 35, 53, 34, -1, 34, 53, 1, -1, 213, 207, 194, -1, 215, 213, 194, -1, 194, 193, 215, -1, 53, 35, 51, -1, 26, 51, 35, -1, 193, 191, 215, -1, 15, 51, 26, -1, 216, 215, 191, -1, 218, 216, 191, -1, 191, 190, 218, -1, 190, 189, 218, -1, 219, 218, 189, -1, 174, 219, 189, -1, 171, 174, 189, -1, 135, 14, 20, -1, 135, 20, 138, -1, 138, 20, 16, -1, 138, 16, 141, -1, 211, 198, 201, -1, 202, 201, 198, -1, 141, 16, 27, -1, 141, 27, 143, -1, 210, 204, 199, -1, 194, 207, 199, -1, 143, 27, 28, -1, 143, 28, 52, -1, 52, 28, 33, -1, 52, 33, 2, -1, 2, 33, 40, -1, 2, 40, 0, -1, 0, 40, 69, -1, 0, 69, 43, -1, 43, 69, 70, -1, 43, 70, 46, -1, 46, 70, 74, -1, 46, 74, 49, -1, 49, 74, 78, -1, 49, 78, 54, -1, 54, 78, 85, -1, 54, 85, 58, -1, 58, 85, 87, -1, 58, 87, 60, -1, 60, 87, 145, -1, 98, 145, 87, -1, 168, 169, 128, -1, 168, 128, 127, -1, 168, 127, 166, -1, 127, 167, 166, -1, 134, 148, 149, -1, 134, 149, 130, -1, 133, 147, 148, -1, 133, 148, 134, -1, 133, 86, 147, -1, 61, 147, 86, -1, 132, 146, 145, -1, 132, 145, 98, -1, 131, 150, 146, -1, 131, 146, 132, -1
            ]
          }
        }
      ]
      name "group_0_2829873"
    }
  ]
  name "pista_Desafio.obj"
}
Transform {
  translation 0.85 0.57 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Cylinder {
        height 0.1
        radius 0.025
      }
    }
  ]
}
Transform {
  translation -0.37 -0.47 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.15 0.01
      }
    }
  ]
}
DEF ROBOT Robot {
  translation 0.791978 0.469404 0.05
  rotation 0 0 1 -2.1435353071795866
  children [
    DEF BODY Transform {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0.364706 0.65098
            metalness 0
          }
          geometry DEF BODY Box {
            size 0.1 0.1 0.07
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.045 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.008908300000000001 0.0550005 -0.0190634
        rotation 0.005239597594134638 -0.005240627593661693 -0.9999725408427353 1.5708153071795863
        children [
          DEF WHEEL Transform {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.034
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.00493567 -0.0550005 -0.015931800000000003
        rotation -0.06235709316051231 0.06235691631645431 -0.9961040146089533 1.5746998877097693
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Transform {
      translation 0.0385 -0.02 0.063
      rotation -0.726828171413359 -0.4856551145357015 -0.48565411453546564 -1.884582
      children [
        DEF UNUSED_EYE Group {
          children [
            Transform {
              translation 0.0348354 -0.0358685 0
              rotation 0 0 1 -0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
          ]
        }
      ]
    }
    Transform {
      translation 0.0385 0.02 0.063
      rotation -0.7268306455324123 0.4856527631523045 0.4856527631523045 -1.884578
      children [
        USE UNUSED_EYE
      ]
    }
    DistanceSensor {
      translation 0.06 0.01 0.02
      rotation 0 1 0 1.5708003061004252
      name "ds0"
      type "infra-red"
      numberOfRays 2
    }
    DistanceSensor {
      translation 0.06 -0.02 0.02
      rotation 0 1 0 1.5708003061004252
      name "ds1"
      type "infra-red"
      numberOfRays 2
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0 0.0415
    rotation 0 0 1 1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "controlador"
}
